src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Mvc/ViewComponentMetadata.cs

```diff
@@ -10,6 +10,7 @@ namespace Microsoft.AspNetCore.Mvc.Razor.Extensions;
 public sealed record ViewComponentMetadata() : MetadataObject(MetadataKind.ViewComponent)
 {
     public required string Name { get; init; }
+    internal TypeNameObject? OriginalTypeName { get; init; }
```

Comment:
FWIW, to follow the same pattern where `TypeNameObject` is used elsewhere, this internal property should be renamed to `OriginalTypeNameObject`. Then, provide a computed property that returns a string:

```suggestion
    internal required TypeNameObject OriginalTypeNameObject { get; init; }
    
    public string OriginalTypeName => OriginalTypeNameObject.FullName.AssumeNotNull();
```

The purpose of the `TypeNameObject` property is for serialization.

Reply:
Just so you know I'm not ignoring your suggestions, it can't be required because of visibility issues.